hspec-core/src/Test/Hspec/Core/Config/Definition.hs

```diff
@@ -353,6 +358,7 @@ runnerOptions = [
   , mkOptionNoArg "rerun-all-on-success" Nothing setRerunAllOnSuccess "run the whole test suite after a previously failing rerun succeeds for the first time (only works in combination with --rerun)"
   , mkOption "jobs" (Just 'j') (argument "N" readMaxJobs setMaxJobs) "run at most N parallelizable tests simultaneously (default: number of available processors)"
   , option "seed" (argument "N" readMaybe setSeed) "used seed for --randomize and QuickCheck properties"
+  , option "max-time-per-test" (argument "N" (fmap Seconds . readMaybe) setMaxTimePerTest) "each test will run at most N seconds"
```

Comment:
What do you think about naming the option `--timeout` instead of `--max-time-per-test`? Would that work for you? 

I think we also want `--no-timeout` to set it back to `Nothing`. Config options [can come from various places](https://hspec.github.io/options.html#precedence-resolution-for-options-that-are-specified-in-multiple-places), so you may want to set it back to nothing at the command line.